src/camera.js

```diff
@@ -303,10 +303,17 @@
 
     /**
      * Getter/setter for the viewport.
+     *
+     * The viewport consists of a width and height in pixels, plus a left and
+     * top offset in pixels.  The offsets are only used to determine if pixel
+     * alignment is possible.
      */
     Object.defineProperty(this, 'viewport', {
       get: function () {
-        return {width: this._viewport.width, height: this._viewport.height};
+        return {
```

Comment:
In theory, it should be more efficient to do this ourselves than use jQuery's extend, since we know what keys are of interest.